admin/spam/views.py

```diff
@@ -174,6 +174,8 @@ def form_valid(self, form):
         try:
             if int(form.cleaned_data.get('confirm')) == Comment.SPAM:
                 item.confirm_spam(save=True)
+                item.is_deleted = True
```

Comment:
The problem is that `SpamMixin` doesn't know what it is going to be mixed in with. Right now it is only comments, which has the `is_deleted`, but if we apply this to other models it might not be. Either we will have to modify those models with a migration, or adjust how it is shown to be removed.